stack.yaml

```diff
@@ -0,0 +1,61 @@
+resolver: lts-5.13
```

Comment:
I wanted to build this repo against the HEAD of each of its dependencies. So the commit IDs here are the latest commit IDs of those repositories (at the time when I was working on this).

Here are some other ways of achieving the same thing:
* Put `../docutils`, `../diagrams-builder`, etc. under the `packages` section of `stack.yaml`. This assumes that the dependencies are on your local machine in those folders, and that the dependencies are up-to-date.
* Write a script that generates the `stack.yaml` file with the latest commit IDs.
* Store the dependences as git submodules.

I think these all have different benefits and drawbacks.

The benefit of the approach I used here is that the code in the repo won't break due to changes in other repos; in other words, if a version of the repo works, it will always work, since the versions of the dependencies that it's built against (indexed by the git commit ID) are "immutable".

Of course, the downside is that it would be very annoying to update this file every time changes are made to one of the dependencies' repo.

What are your thoughts on this?